File: app/src/main/java/cryptography/biometric/di/ViewModelFactoryModule.java
package cryptography.biometric.di;

import androidx.lifecycle.ViewModelProvider;

import cryptography.biometric.viewmodels.ViewModelProviderFactory;
import dagger.Binds;
import dagger.Module;


@Module
public abstract class ViewModelFactoryModule {
    @Binds
    public abstract ViewModelProvider.Factory bindViewModelFactory(ViewModelProviderFactory viewModelFactory);
}

File: app/src/main/java/cryptography/biometric/di/ViewModelFactoryModule.kt
package cryptography.biometric.di

import androidx.lifecycle.ViewModelProvider
import cryptography.biometric.viewmodels.ViewModelProviderFactory
import dagger.Binds
import dagger.Module

/**
 * Provides the object for ViewModelProviderFactory
 */
@Module
abstract class ViewModelFactoryModule {
    /**
     * Binds also create object similar to Provide only difference is when we dont have method body use Binds
     * Binds create object of ViewModelProviderFactory
     */
    @Binds
    abstract fun bindViewModelFactory(viewModelFactory: ViewModelProviderFactory): ViewModelProvider.Factory
}